public/app/plugins/datasource/dashboard/DashboardQueryEditor.tsx

```diff
@@ -141,7 +142,7 @@ export class DashboardQueryEditor extends PureComponent<Props, State> {
         continue;
       }
 
-      if (panel.targets && panel.datasource !== SHARED_DASHBODARD_QUERY) {
+      if (panel.targets && panel.id !== dashboard.panelInEdit?.id) {
```

Comment:
I think you need to re add the existing filter. If understand the old version of the condition correctly it will filter out any panels that are using the `SHARED_DASHBODARD_QUERY` data source and I think we still want that filter and the new one that you added, right?

```suggestion
      if (panel.targets && panel.id !== dashboard.panelInEdit?.id && panel.datasource.uid !== SHARED_DASHBOARD_QUERY) {
```

Reply:
I think this `render` function looks kind of insane too, but we should refactor that in a separate PR and focus on just fixing the issue.

Reply:
ah good catch 🙌